lib/storage.py

```diff
@@ -177,6 +177,9 @@ def write(self):
             self._write()
 
     def _write(self):
+        if not self.path:
```

Comment:
It isn't enough just for this check to succeed.  I can't figure out how to use this class and not get it to crash for anything other than constructing this as if it were a fancy dict.

If you give an instance of this with `self.path == None` to eg a `Wallet` object, it will crash on `__init__`.

Would you mind finishing this PR?